sky/cloud_stores.py

```diff
@@ -122,6 +154,7 @@ def is_directory(self, url: str) -> bool:
         commands = [self._INSTALL_GSUTIL]
         commands.append(f'{self._gsutil_command} ls -d {url}')
         command = ' && '.join(commands)
+        self.check_path_exists(url)
```

Comment:
Instead of adding this, maybe we can simply check the command output below and see if it contains `CommandException: One or more URLs matched no objects.`. That should be sufficient to raise the error.

```
(base) ➜  ~ gsutil ls -d gs://romilb-notebooks/my_datasets
CommandException: One or more URLs matched no objects.
(base) ➜  ~ gsutil ls -d gs://romilb-nonexistentbucket 
BucketNotFoundException: 404 gs://romilb-nonexistentbucket bucket does not exist.
```